homeassistant/components/renault/renault_hub.py

```diff
@@ -116,3 +121,11 @@ async def get_account_ids(self) -> list[str]:
     def vehicles(self) -> dict[str, RenaultVehicleProxy]:
         """Get list of vehicles."""
         return self._vehicles
+
+
+async def with_error_wrapping(action: Coroutine[Any, Any, _T]) -> _T:
+    """Run action list of vehicles."""
+    try:
+        return await action
+    except RenaultException as err:
+        raise HomeAssistantError(err) from err
```

Comment:
Consider making this a decorator instead.

Reply:
Decorator added in #86071
I will rebase here once it is merged.

Reply:
Merged/Rebased